core/src/main/scala/quasar/destination/avalanche/AvalancheTransactorConfig.scala

```diff
@@ -16,39 +16,146 @@
+
+  def transactorConfig[F[_]: ConcurrentEffect: Timer: ContextShift](
+      getAuth: GetAuth[F])
+      : F[Either[String, TransactorConfig]] = {
+
+    auth match {
+      case UsernamePassword(username, password) =>
+        AvalancheTransactorConfig.fromUsernamePassword(connectionUri, username, password).asRight[String].pure[F]
+
+      case ExternalAuth(authId, userinfoUri, userinfoField) =>
+        (
+          for {
+            token <- EitherT(getToken[F](getAuth, authId))
+            email <- EitherT.fromOptionF(
+              UserInfoGetter.emailFromUserinfo(token, userinfoUri, userinfoField),
+              "Querying user info using the token acquired via the auth key did not yield an email. Check the scopes granted to the token.")
+          } yield AvalancheTransactorConfig.fromToken(connectionUri, Username(email.asString), token)
+        ).value
+
+    }
+  }
+}
 
 object AvalancheTransactorConfig {
   val IngresDriverFqcn: String = "com.ingres.jdbc.IngresDriver"
   // Avalanche closes the connection after 4 minutes so we set a connection lifetime of 3 minutes.
   val MaxLifetime: FiniteDuration = 3.minutes
   val PoolSize: Int = 8
 
-  def apply(
+  private[this] def appendJdbcParamString(jdbcUri: URI, params: String): URI = {
+      val u = jdbcUri.toString
+
+      if (u.endsWith(";"))
+        URI.create(u + params)
+      else
+        URI.create(u + ";" + params)
+  }
+
+  private[this] def fromDetails(
       connectionUrl: URI,
       username: Username,
-      password: ClusterPassword)
+      password: String)
       : TransactorConfig = {
 
-    val fullUrl = {
-      val u = connectionUrl.toString
-      val auth = s"UID=${username.asString};PWD=${password.asString}"
-
-      if (u.endsWith(";"))
-        URI.create(u + auth)
-      else
-        URI.create(u + ";" + auth)
-    }
+    val fullUrl = appendJdbcParamString(connectionUrl, s"UID=${username.asString};PWD=$password")
 
     val driverConfig =
       JdbcDriverConfig.JdbcDriverManagerConfig(fullUrl, Some(IngresDriverFqcn))
 
     TransactorConfig(driverConfig, None)
   }
+
+  def fromUsernamePassword(
+      connectionUrl: URI,
+      username: Username,
+      password: ClusterPassword)
+      : TransactorConfig = 
+    fromDetails(connectionUrl, username, password.asString) 
+
+  private val utf8 = Charset.forName("UTF-8")
+
+  def fromToken(
+      connectionUrl: URI,
+      username: Username,
+      token: Credentials.Token)
+      : TransactorConfig = {
+
+    // for token auth the auth_type must be set to "browser"
+    // If any 'auth_type' is already set we don't override,
+    // but we do append `auth_type=browser' if it is missing
+    val fullUri = 
+      if (connectionUrl.toString.matches(raw".*(/|;)auth_type=[^;]+.*"))
+        connectionUrl 
+      else 
+        appendJdbcParamString(connectionUrl, "auth_type=browser")
+
+    fromDetails(
+      fullUri, 
+      username, 
+      s"access_token=${new String(token.toByteArray, utf8)}")
```

Comment:
Just double checking: so the password field becomes something like `PWD=access_token=<token>`? Maybe worth adding a small test case for this (and the `auth_type` logic above)?

Reply:
You are exactly correct.

I'll add the test case 👍